### PR TITLE
WCAndroid: Update Fastfile with version model

### PR DESCRIPTION
**Pull request by spencertransier**
## Summary
- This PR integrates the new version and build code models from https://github.com/wordpress-mobile/release-toolkit/pull/512
- It currently points to a development version of the Release Toolkit so that I could test out the call site implementation before releasing a new version of the Release Toolkit. 
- Nearly all of the `precheck` actions from the Release Toolkit were tightly coupled with the existing version methods. To simplify things with the new version model, I took the relevant snippets from the precheck actions and added them directly here. This is something that I've discussed with Olivier and Jeremy before in the context of "no one knows what the precheck actions do unless you dive deep down into the RT". These changes are somewhat opposite of DRY, but I think it's worth it for transparency. We may also decide in the future to create smaller actions that do more specific things versus the existing precheck and version bump actions that combine many disparate methods. 

@jkmassel I won't merge this until after the 15.6 code freeze begins, just to make sure any unforeseen issues don't interfere with that. 

## Before Merging
- [x] Point the Release Toolkit to a stable version
- [x] WCAndroid versions must be switched from `build.gradle` to `version.properties` (Separate PR: https://github.com/woocommerce/woocommerce-android/pull/9866)

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -105,10 +109,43 @@ platform :android do
...
+
+     Do you want to continue?
+    MESSAGE
+
+    # Ask user confirmation
+    UI.user_error!('Aborted by user request') unless options[:skip_confirm] || UI.confirm(confirmation_message)
+
+    # Create the release branch
+    UI.message 'Creating release branch...'
+    ensure_git_status_clean
```
Comment (jkmassel) [inline fastlane/Fastfile]: We should do this at the top of the lane – it's not a great experience if we give input then find out that the starting parameters were invalid

File: Gemfile [hunk 2]
```diff
@@ -8,9 +8,9 @@ gem 'rubocop', '~> 1.56'
 
 ### Fastlane Plugins
 
-gem 'fastlane-plugin-wpmreleasetoolkit', '~> 9.0'
+# gem 'fastlane-plugin-wpmreleasetoolkit', '~> 9.0'
 # gem 'fastlane-plugin-wpmreleasetoolkit', path: '../../release-toolkit'
-# gem 'fastlane-plugin-wpmreleasetoolkit', git: 'https://github.com/wordpress-mobile/release-toolkit', branch: '…'
+ gem 'fastlane-plugin-wpmreleasetoolkit', git: 'https://github.com/wordpress-mobile/release-toolkit', branch: 'add/version-model'
```
Comment (jkmassel) [inline Gemfile]: We'll need to use a specific version before landing this

**Review by jkmassel — CHANGES_REQUESTED** [review]
I'm getting this error:

> [14:51:13]: Exit status of command 'git add ./WooCommerce/build.gradle' was 128 instead of 0.
warning: could not open directory 'fastlane/WooCommerce/': No such file or directory
fatal: pathspec './WooCommerce/build.gradle' did not match any files

When trying to run the code freeze lane

**Comment by spencertransier** [comment]
> I'm getting this error:
> 
> [14:51:13]: Exit status of command 'git add ./WooCommerce/build.gradle' was 128 instead of 0.
> warning: could not open directory 'fastlane/WooCommerce/': No such file or directory
> fatal: pathspec './WooCommerce/build.gradle' did not match any files
> 
> When trying to run the code freeze lane

@jkmassel Is it trying to switch to `trunk` and running into issues? I had to disable the branch switches to be able to test the lanes: 
Screenshot 2023-09-29 at 4 10 39 PM

File: fastlane/Fastfile (hunk 1 above)
Comment (spencertransier) [reply fastlane/Fastfile]: Addressed in https://github.com/woocommerce/woocommerce-android/commit/25d15be1c32d65df177f08905062219fba7a0915

**Review by jkmassel — APPROVED** [review]

**Pull request merged**